src/features/payment/components/IframeCardForm/IframeCardForm.tsx

```diff
@@ -0,0 +1,289 @@
+          //   HF0001 -generic build field error
+          //   HF0002 -temporary unavailability of the service
+          //   HF0003 -session expired–the payment experience shall be restarted from the post orders/build
+          //   HF0004 -card validation error–the key check on the card number was failed
+          //   HF0005 -brand not found–the card brand is not in the list of supported brands
+          //   HF0006 -expiration date exceeded–the provided card is expired
+          //   HF0007 –internal error –if the issue persists the payment has to be restarted
+          //   HF0009 –3DS GDI verification failed –the payment experience has to be stopped with failure.
+          const { id, errorCode, errorMessage } = evtData;
+          fieldFormStatus.set(id, {
+            isValid: false,
+            errorCode,
+            errorMessage,
+          });
+          formStatus = calculateFormValidStatus(fieldFormStatus);
+          setDummyState(Math.random);
+        },
+        onConfirmError(evtData: any) {
+          // this event is returned as a consequence of the invocation of confirmData() SDK function.
+          // the possible cases are:
+          //   HF0002 –temporary unavailability of the service
+          //   HF0003 -session expired–the payment experience shall be restarted from the post orders/build
+          //   HF0007 –internal error–if the issue persists the payment has to be restarted
+          // eslint-disable-next-line no-console
+          console.log("onConfirmError", evtData);
+        },
+        onBuildFlowStateChange(
+          evtData: any,
+          state:
+            | "READY_FOR_PAYMENT"
+            | "REDIRECTED_TO_EXTERNAL_DOMAIN"
+            | "PAYMENT_COMPLETE"
+        ) {
+          // this event is returned for each state transition of the payment state machine.
+          // the possible states expressed by the value state are:
+          // READY_FOR_PAYMENT: the card data has been properly collected and it is now possible to
+          //   invoke the server to server
+          //   posthttps://{nexiDomain}/api/phoenix-0.0/psp/api/v1/build/cardData?sessionId={thesessionId}
+          //   to collect the non-PCI card information.
+          // REDIRECTED_TO_EXTERNAL_DOMAIN: when this state is provided, the browser has to be redirected to
+          //   the evtData.data.url external domain for strong customer authentication (i.e ACS URL).
+          // PAYMENT_COMPLETE: the payment experience is finished. It is required to invoke
+          //   the get https://{nexiDomain}/api/phoenix-0.0/psp/api/v1/build/state?sessionId={thesessionId}  },
+          // eslint-disable-next-line no-console
+          console.log("onBuildFlowStateChange", evtData, state);
+          if (state === "READY_FOR_PAYMENT") {
+            void (async () => {
+              try {
+                const response = await fetch(
+                  `/ecommerce/checkout/v1/payment-methods/${form.sessionId}`,
```

Comment:
Same for the logic to be moved. As for openapi it will be updated the PR https://github.com/pagopa/pagopa-infra/pull/1241 to be able to generate the client as soon as possible 🙂

Reply:
pay attention,  we want to invoke this api before creating the transaction on eCommerce? For now we can leave PR as is, and address the problem in task CHK-1822